### PR TITLE
oci registry

### PR DESCRIPTION
**Pull request by chGoodchild**
Replacing chGoodchild:main_oci_docker_registry

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -101,9 +101,18 @@ jobs:
...
           $powerloader_path download -f test/remote_mirrors.yml -d $server_path
+
+          # Upload file to AWS mock
           aws --endpoint-url $AWS_S3_ENDPOINT s3 cp $server_path/xtensor-0.23.9-hc021e02_1.tar.bz2 s3://testbucket/
           # aws --endpoint-url $AWS_S3_ENDPOINT s3 ls s3://testbucket
 
+          # Upload file to OCI mock: https://docs.docker.com/engine/reference/commandline/tag/
+          mkdir test/tmp
+          echo "artifact" > test/tmp/artifact
+          ./build/powerloader upload /tmp/artifact:artifact:321 -m oci://localhost:5000
```
Comment (wolfv) [inline .github/workflows/main.yml:112]: wrong path

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -101,9 +101,18 @@ jobs:
...
+
+          oras push localhost:5000/artifact:v1 --manifest-config \
+                                  /dev/null:application/vnd.acme.rocket.config ./artifact
```
Comment (wolfv) [inline .github/workflows/main.yml:115]: wrong path

**Comment by chGoodchild** [comment]
Download still doesn't work...

**Comment by wolfv** [comment]
yeah you need `-k --plain-http` probably

**Comment by chGoodchild** [comment]
`test_OCImain_branch` won't work yet, because I made some simplifications in `helpers.py`. Will fix it soon...

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -109,7 +109,7 @@ jobs:
           # Upload file to OCI mock: https://docs.docker.com/engine/reference/commandline/tag/
           mkdir test/tmp
           echo "artifact" > test/tmp/artifact
-          ./build/powerloader upload test/tmp/artifact:artifact:321 -m oci://localhost:5000 -k -v --plain-http
+          ./build/powerloader upload test/tmp/artifact:artifact:1.0 -m oci://localhost:5000 -k -v --plain-http
 
           oras push localhost:5000/artifact:1.0 --manifest-config \
```
Comment (wolfv) [inline .github/workflows/main.yml:114]: do we need this?

Comment (chGoodchild) [reply .github/workflows/main.yml:114]: Good point, we don't need it anymore...

**Comment by chGoodchild** [comment]
Some tests are still not ready

**Comment by wolfv** [comment]
OK, but looks like your on a good trajectory :)

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -101,9 +101,16 @@ jobs:
...
           $powerloader_path download -f test/remote_mirrors.yml -d $server_path
+
+          # Upload file to AWS mock
           aws --endpoint-url $AWS_S3_ENDPOINT s3 cp $server_path/xtensor-0.23.9-hc021e02_1.tar.bz2 s3://testbucket/
           # aws --endpoint-url $AWS_S3_ENDPOINT s3 ls s3://testbucket
 
+          # Upload file to OCI mock: https://docs.docker.com/engine/reference/commandline/tag/
+          mkdir test/tmp
+          echo "artifact" > test/tmp/mock_artifact
+          oras push localhost:5000/mock_artifact:1.0 --manifest-config \
```
Comment (wolfv) [inline .github/workflows/main.yml:112]: I don't understand how this is supposed to work? Where does `oras` come from?

Comment (wolfv) [reply .github/workflows/main.yml:112]: In fact it looks like it's pre-installed. Nice! https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md#cli-tools

Comment (chGoodchild) [reply .github/workflows/main.yml:114]: Good point, I already had it set up on my machine. Perhaps the line below would help?
https://github.com/mamba-org/powerloader/blob/25e005fd9712e2f2c59980303112d62767405fd5/.github/workflows/main.yml#L92

File: test/test_OCImock.py [hunk 5]
```diff
@@ -12,38 +9,34 @@ def setup_class(cls):
...
-        username = self.set_username()
+        # Upload
         upload_path = generate_unique_file(file)
         hash_before_upload = calculate_sha256(upload_path)
-        tag, name_on_server = upload_oci(upload_path, powerloader_binary, file["oci_upload_location"])
+        tag, name_on_server = upload_oci(upload_path, powerloader_binary, file["oci_mock_server"], plain_http=True)
         Path(upload_path).unlink()
-        newpath, tmp_yaml = generate_oci_download_yml(file, tag, name_on_server, username)
-        download_oci_file(powerloader_binary, tmp_yaml, file)
+
```
Comment (wolfv) [inline test/test_OCImock.py:36]: Can you check `api/v2/...` that it contains the tag and image?

Comment (wolfv) [reply test/test_OCImock.py:36]: You can use `requests` library to make some simple HTTP requests.

File: .github/workflows/main.yml (hunk 4 above)
Comment (wolfv) [reply .github/workflows/main.yml:114]: no, we don't need it :)

File: test/test_OCImock.py (hunk 5 above)
Comment (chGoodchild) [reply test/test_OCImock.py:36]: Yes, good idea :)

File: .github/workflows/main.yml (hunk 4 above)
Comment (chGoodchild) [reply .github/workflows/main.yml:114]: aah, because I can use powerloader to upload the artifact :)

File: test/test_OCImock.py (hunk 5 above)
Comment (chGoodchild) [reply test/test_OCImock.py:36]: Like this perhaps?
https://github.com/mamba-org/powerloader/blob/41dbc5109791b5689f7e09bea4cfb1455ebbafca/test/helpers.py#L144-L155

Comment (wolfv) [reply test/test_OCImock.py:36]: yeah, does it work?

**Pull request merged**